From b6a85b9272ab9638d7fd61e309e2c6469d4751d8 Mon Sep 17 00:00:00 2001
From: Erik Larsson <who+github@cnackers.org>
Date: Sun, 31 Jan 2021 11:13:30 +0100
Subject: [PATCH] python-uci: add package

Python bindings for libuci, useful for writing ansible modules to manage OpenWrt systems.
Reused Makefile from upstream:
https://gitlab.nic.cz/turris/turris-os-packages/-/blob/d8a430fc7b67d82d57537b4904d3769ee460aa4b/lang/python-uci/Makefile

Signed-off-by: Erik Larsson <who+github@cnackers.org>
---
 lang/python/python-uci/Makefile | 40 +++++++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)
 create mode 100644 lang/python/python-uci/Makefile

diff --git a/lang/python/python-uci/Makefile b/lang/python/python-uci/Makefile
new file mode 100644
index 000000000..67807fed5
--- /dev/null
+++ b/lang/python/python-uci/Makefile
@@ -0,0 +1,40 @@
+#
+# Copyright (C) 2018-2020 CZ.NIC, z. s. p. o. (https://www.nic.cz/)
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=python-uci
+PKG_VERSION:=0.8.1
+PKG_RELEASE:=1
+
+PYPI_NAME:=pyuci
+PKG_HASH:=9287fe41b427dc5c167592d429be48c1e6cfe276225681b1bdefddfe90d7e941
+
+PKG_MAINTAINER:=Josef Schlehofer <josef.schlehofer@nic.cz>
+PKG_LICENSE:=GPL-3.0-or-later
+PKG_LICENSE_FILES:=LICENSE
+
+include ../pypi.mk
+include $(INCLUDE_DIR)/package.mk
+include ../python3-package.mk
+
+define Package/python3-uci
+  SECTION:=lang
+  CATEGORY:=Languages
+  SUBMENU:=Python
+  TITLE:=Python UCI bindings
+  URL:=https://gitlab.nic.cz/turris/pyuci/
+  DEPENDS:=+python3-light +libuci
+  VARIANT:=python3
+endef
+
+define Package/python3-uci/description
+  Python3 bindings for Unified Configuration Interface.
+endef
+
+$(eval $(call Py3Package,python3-uci))
+$(eval $(call BuildPackage,python3-uci))
+$(eval $(call BuildPackage,python3-uci-src))
--
2.25.1
